### DISCUSSION TITLE
Login session duration

### DISCUSSION THREAD

**Discussion by AndreasNasman** [ajrosen/Bitwarden-Accelerator#16] [Q&A] [answered]
Hi! 👋
Thanks for this excellent Alfred Workflow! ❤️

I'm unsure if this question relates to this project or the Bitwarden CLI, but is there a way to configure the duration of your login session? From what I've tested, it seems to be one day. If possible, I'd like to customize this so that the login (and unlock) remain active longer. Is this possible to configure in the workflow or in the Bitwarden CLI?

**Comment by ajrosen** [comment, marked as answer]
I'm glad you like it!

There's no session timeout, but I think sometimes the **_bw serve_** process terminates (crashes).  If/when it does you're logged out of Bitwarden.  The workflow uses it as a cache, and as a proxy for Bitwarden's API.  It doesn't have any options.

I can _add_ a timeout to the workflow.  It wouldn't help in your case though, unless Bitwarden releases a version that never exits unexpectedly.  It would effectively be "logout after N minutes or when _bw serve_ crashes, whichever happens first".

**Reply by AndreasNasman** [reply]
Thanks for the reply and explanation! I'll keep an eye on that process in the future; I might see a pattern as to when it crashed. There is no need to add a timeout on my part, then. 👍
